src/github/mod.rs

```diff
@@ -217,4 +255,37 @@ mod tests {
         let path = "path/to/file.tsx";
         assert!(should_index(path));
     }
+
+    #[tokio::test]
+    async fn test_is_indexing_allowed() {
+        // Permissible
+        let repository = Repository {
+            owner: "open-sauced".to_string(),
+            name: "ai".to_string(),
+            branch: "beta".to_string(),
+        };
+
+        let is_allowed = is_indexing_allowed(&repository).await.unwrap_or_default();
+        assert_eq!(is_allowed, true);
+
+        //Permissible
+        let repository = Repository {
+            owner: "facebook".to_string(),
+            name: "react".to_string(),
+            branch: "main".to_string(),
+        };
+
+        let is_allowed = is_indexing_allowed(&repository).await.unwrap_or_default();
+        assert_eq!(is_allowed, true);
+
+        //Imermissible
```

Comment:
nit: Spelling. Should be `Impermissible`

```suggestion
        //Impermissible
```